cmd/oras/root/push.go

```diff
@@ -182,6 +192,22 @@ func runPush(cmd *cobra.Command, opts *pushOptions) error {
 		}
 		desc.Annotations = packOpts.ConfigAnnotations
 		packOpts.ConfigDescriptor = &desc
+	} else if opts.Platform.Platform != nil {
+		blob, err := json.Marshal(opts.Platform.Platform)
+		if err != nil {
+			return err
+		}
+		mediaType := opts.artifactType
+		if mediaType == "" {
+			mediaType = oras.MediaTypeUnknownConfig
+		}
```

Comment:
This logic works for OCI spec v1.0. However, it does not work very well for OCI spec v1.1. And I agree with you w.r.t. https://github.com/oras-project/oras/pull/1500#discussion_r1801538753

Reply:
Your original code looks better to me
```go
mediaType := oras.MediaTypeUnknownConfig
if opts.Flag == option.ImageSpecV1_0 && opts.artifactType != "" {
	mediaType = opts.artifactType
}
```

Reply:
The original code was clearer at least.

Reply:
Can you apply your original code above to this PR?